addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComInterfaceMessage.java

```diff
@@ -257,8 +301,27 @@ public void encodeMessage(byte[] data) throws RFXComException {
             enableARCPackets = (data[9] & 0x02) != 0;
             enableX10Packets = (data[9] & 0x01) != 0;
 
-            hardwareVersion1 = data[10];
-            hardwareVersion2 = data[11];
+            /**
+             * Different firmware versions have slightly different message formats.
+             * The firmware version numbering is unique to each hardware version
+             * but the location of the hardware version in the message is one of
+             * those things whose position varies. So we have to just look at the
+             * firmware version and pray. This condition below is taken from the
+             * openhab1-addons binding.
```

Comment:
@mjagdis can it be that we also have to port the actual updated handling of the firmware version, I just noted this line in a test:

https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.rfxcom.test/src/test/java/org/openhab/binding/rfxcom/internal/messages/RFXComInterfaceMessageTest.java#L62